apps/api/src/app/integrations/usecases/get-active-integration/get-active-integration.usecase.ts

```diff
@@ -59,7 +74,12 @@ export class GetActiveIntegrations {
     selectedIntegrations: IntegrationEntity[]
   ): GetActiveIntegrationResponseDto[] {
     return activeIntegration.map((integration) => {
-      const selected = selectedIntegrations.find((selectedIntegration) => selectedIntegration._id === integration._id);
+      // novu integrations doesn't have unique id that's why we need to compare by environmentId
+      const selected = selectedIntegrations.find(
+        (selectedIntegration) =>
+          selectedIntegration._id === integration._id &&
+          selectedIntegration._environmentId === integration._environmentId
+      );
```

Comment:
fixed a bug: `the endpoint marks few integrations for the same channel for ex. email as selected`, like it was basically marking the novu-email as selected when it was returned by the GetDecryptedIntegrations use-case...